src/llscan.cc

```diff
@@ -0,0 +1,470 @@
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include <algorithm>
+#include <fstream>
+#include <vector>
+
+#include <lldb/API/SBExpressionOptions.h>
+
+#include "src/llnode.h"
+#include "src/llv8.h"
+#include "src/llv8-inl.h"
+#include "src/llscan.h"
+
+namespace llnode {
+
+// Defined in llnode.cc
+extern v8::LLV8 llv8;
+
+struct MemoryRange {
+    uint64_t start;
+    uint64_t length;
+    MemoryRange* next;
+};
+
+static MemoryRange* ranges = nullptr;
+
+std::map<uint64_t, TypeRecord*>* mapstoinstances = NULL;
+
+using namespace lldb;
+
+bool ListObjectsCmd::DoExecute(SBDebugger d, char** cmd,
+        SBCommandReturnObject& result) {
+
+    SBTarget target = d.GetSelectedTarget();
+    if (!target.IsValid()) {
+        result.SetError("No valid process, please start something\n");
+        return false;
+    }
+
+    /* Ensure we have a map of objects. */
+    if (!ScanHeapForObjects(&target, result)) {
+        return false;
+    }
+
+    /* Create a vector to hold the entries sorted by instance count
+     * TODO - Make sort type an option (by count, size or name)
+     */
+    std::vector<TypeRecord*> sorted_by_count = std::vector<TypeRecord*>();
+    for (std::map<uint64_t, TypeRecord*>::iterator it = mapstoinstances->begin();
+            it != mapstoinstances->end(); ++it) {
+        sorted_by_count.push_back(it->second);
+    }
+
+    std::sort(sorted_by_count.begin(), sorted_by_count.end(),
+            TypeRecord::compareInstanceCounts);
+
+    uint64_t total_objects = 0;
+
+    result.Printf( "Map                Instances  Total Size Name\n" );
+    result.Printf( "------------------ ---------- ---------- ----\n" );
+
+    for (std::vector<TypeRecord*>::iterator it = sorted_by_count.begin();
+            it != sorted_by_count.end(); ++it) {
+        TypeRecord* t = *it;
+        result.Printf("0x%016" PRIx64 " %10" PRId64 " %10" PRId64 " %s\n",
+                t->map, t->instance_count, t->total_instance_size,
+                t->type_name->c_str());
+        total_objects += t->instance_count;
+    }
+
+    return true;
+}
+
+/* Utility function to generate short type names for objects.
+ * Based on the inspection code in llv8.cc
+ * TODO Merge back in to llv8.cc?
+ */
+std::string GetTypeName( llnode::v8::HeapObject *heap_object, v8::Value::InspectOptions *options, v8::Error err ) {
+  int64_t type = heap_object->GetType(err);
+  v8::LLV8 *v8 = heap_object->v8();
+  if (type == v8->types()->kGlobalObjectType) return heap_object->Inspect(options, err);
+  if (type == v8->types()->kCodeType) return heap_object->Inspect(options, err);
+  if (type == v8->types()->kMapType) {
+    return heap_object->Inspect(options, err);
+  }
+
+  if (type == v8->types()->kJSObjectType) {
+
+      llnode::v8::HeapObject map_obj = heap_object->GetMap(err);
+      if (err.Fail()) {
+          return std::string();
+      }
+
+      llnode::v8::Map map(map_obj);
```

Comment:
I don't think that `llnode::` is need here or anywhere else, all functions are in `llnode` namespace already.